### PR TITLE
Add DescribeAlias and ListAliases interfaces

### PR DESCRIPTION
**Pull request by wayblink**
related: #22882

### CODE REVIEW

**Comment by wayblink** [comment]
rerun ut

File: internal/rootcoord/root_coord.go [hunk 1]
```diff
@@ -1653,6 +1658,79 @@ func (c *Core) AlterAlias(ctx context.Context, in *milvuspb.AlterAliasRequest) (
...
+	log.Ctx(ctx).Info("received request to describe alias",
+		zap.String("role", typeutil.RootCoordRole),
+		zap.String("db", in.GetDbName()),
+		zap.String("alias", in.GetAlias()))
+
+	collectionName, err := c.meta.DescribeAlias(ctx, in.GetAlias(), 0)
+
+	if err != nil {
+		return &milvuspb.DescribeAliasResponse{
+			Status: merr.Status(err),
```
Comment (xiaofan-luan) [inline internal/rootcoord/root_coord.go:1681]: Error code

File: internal/rootcoord/root_coord.go [hunk 2]
```diff
@@ -1653,6 +1658,79 @@ func (c *Core) AlterAlias(ctx context.Context, in *milvuspb.AlterAliasRequest) (
...
+	tr := timerecord.NewTimeRecorder(ListAliases)
+
+	log.Ctx(ctx).Info("received request to list aliases",
+		zap.String("role", typeutil.RootCoordRole),
+		zap.String("db", in.GetDbName()))
+
+	aliases, err := c.meta.ListAliases(ctx, 0)
+
+	if err != nil {
+		return &milvuspb.ListAliasesResponse{
```
Comment (xiaofan-luan) [inline internal/rootcoord/root_coord.go:1717]: error code

File: internal/types/types.go [hunk 3]
```diff
@@ -1245,6 +1265,27 @@ type ProxyComponent interface {
...
+	//
+	// ctx is the context to control request deadline and cancellation
+	// req contains the request params, including collection name and new alias
+	//
+	// The `ErrorCode` of `Status` is `Success` if alter alias successfully;
+	// otherwise, the `ErrorCode` of `Status` will be `Error`, and the `Reason` of `Status` will record the fail cause.
+	// error is always nil
+	DescribeAlias(ctx context.Context, req *milvuspb.DescribeAliasRequest) (*milvuspb.DescribeAliasResponse, error)
+
+	// ListAliases query RootCoord to show all alias for the db
```
Comment (xiaofan-luan) [inline internal/types/types.go:1279]: should we support alias for single collection?

File: internal/types/types.go [hunk 4]
```diff
@@ -1245,6 +1265,27 @@ type ProxyComponent interface {
...
+	//
+	// ctx is the context to control request deadline and cancellation
+	// req contains the request params, including collection name and new alias
+	//
+	// The `ErrorCode` of `Status` is `Success` if alter alias successfully;
+	// otherwise, the `ErrorCode` of `Status` will be `Error`, and the `Reason` of `Status` will record the fail cause.
+	// error is always nil
+	ListAliases(ctx context.Context, req *milvuspb.ListAliasRequest) (*milvuspb.ListAliasesResponse, error)
```
Comment (xiaofan-luan) [inline internal/types/types.go:1287]: Alias need to be cotest with collection rename feature to make sure they can work together

File: internal/distributed/proxy/service.go [hunk 5]
```diff
@@ -711,6 +711,16 @@ func (s *Server) AlterAlias(ctx context.Context, request *milvuspb.AlterAliasReq
 	return s.proxy.AlterAlias(ctx, request)
 }
 
+// DescribeAlias show the alias-collection relation for the specified alias.
+func (s *Server) DescribeAlias(ctx context.Context, request *milvuspb.DescribeAliasRequest) (*milvuspb.DescribeAliasResponse, error) {
+	return s.proxy.DescribeAlias(ctx, request)
+}
+
+// ListAliases list all the alias for the specified db, collection.
+func (s *Server) ListAliases(ctx context.Context, request *milvuspb.ListAliasRequest) (*milvuspb.ListAliasesResponse, error) {
```
Comment (xiaofan-luan) [inline internal/distributed/proxy/service.go:720]: ListAlias should support list alias on one collection?

File: internal/proxy/impl.go [hunk 6]
```diff
@@ -2884,6 +2884,157 @@ func (node *Proxy) AlterAlias(ctx context.Context, request *milvuspb.AlterAliasR
 	return aat.result, nil
 }
 
+// DescribeAlias describe alias of collection.
+func (node *Proxy) DescribeAlias(ctx context.Context, request *milvuspb.DescribeAliasRequest) (*milvuspb.DescribeAliasResponse, error) {
+	if !node.checkHealthy() {
+		return &milvuspb.DescribeAliasResponse{
+			Status: unhealthyStatus(),
```
Comment (xiaofan-luan) [inline internal/proxy/impl.go:2891]: need to return with error code

File: internal/proxy/impl.go [hunk 7]
```diff
@@ -2884,6 +2884,157 @@ func (node *Proxy) AlterAlias(ctx context.Context, request *milvuspb.AlterAliasR
...
+
+	if err := node.sched.ddQueue.Enqueue(lat); err != nil {
+		log.Warn(
+			rpcFailedToEnqueue(method),
+			zap.Error(err))
+		metrics.ProxyFunctionCall.WithLabelValues(strconv.FormatInt(node.session.ServerID, 10), method, metrics.AbandonLabel).Inc()
+
+		return &milvuspb.ListAliasesResponse{
+			Status: &commonpb.Status{
+				ErrorCode: commonpb.ErrorCode_UnexpectedError,
```
Comment (xiaofan-luan) [inline internal/proxy/impl.go:3000]: need to return with error code here

**Comment by wayblink** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

File: internal/rootcoord/meta_table.go [hunk 8]
```diff
@@ -697,6 +700,47 @@ func (mt *MetaTable) AlterAlias(ctx context.Context, alias string, collectionNam
...
 }
 
+func (mt *MetaTable) DescribeAlias(ctx context.Context, alias string, ts Timestamp) (string, error) {
+	mt.ddLock.Lock()
+	defer mt.ddLock.Unlock()
+
+	// check if alias exists.
+	collectionID, ok := mt.collAlias2ID[alias]
+	if !ok {
+		return "", merr.WrapErrAliasNotFound(alias)
```
Comment (longjiquan) [inline internal/rootcoord/meta_table.go:710]: Please add ut for this case.

File: internal/rootcoord/meta_table.go [hunk 9]
```diff
@@ -697,6 +700,47 @@ func (mt *MetaTable) AlterAlias(ctx context.Context, alias string, collectionNam
 	return nil
 }
 
+func (mt *MetaTable) DescribeAlias(ctx context.Context, alias string, ts Timestamp) (string, error) {
+	mt.ddLock.Lock()
```
Comment (longjiquan) [inline internal/rootcoord/meta_table.go:704]: Use `RLock` instead.

File: internal/rootcoord/meta_table.go [hunk 10]
```diff
@@ -697,6 +700,47 @@ func (mt *MetaTable) AlterAlias(ctx context.Context, alias string, collectionNam
...
+
+	// check if alias exists.
+	collectionID, ok := mt.collAlias2ID[alias]
+	if !ok {
+		return "", merr.WrapErrAliasNotFound(alias)
+	}
+
+	collectionMeta, ok := mt.collID2Meta[collectionID]
+	if !ok {
+		return "", merr.WrapErrCollectionIDOfAliasNotFound(collectionID)
```
Comment (longjiquan) [inline internal/rootcoord/meta_table.go:715]: Please add ut for this case.

File: internal/rootcoord/meta_table.go [hunk 11]
```diff
@@ -697,6 +700,47 @@ func (mt *MetaTable) AlterAlias(ctx context.Context, alias string, collectionNam
...
+	}
+
+	return collectionMeta.Name, nil
```
Comment (longjiquan) [inline internal/rootcoord/meta_table.go:718]: Please check if the collection was availiable.

File: internal/rootcoord/meta_table.go [hunk 12]
```diff
@@ -697,6 +700,47 @@ func (mt *MetaTable) AlterAlias(ctx context.Context, alias string, collectionNam
...
+}
+
+func (mt *MetaTable) ListAliases(ctx context.Context, collectionName string, ts Timestamp) ([]string, error) {
+	mt.ddLock.Lock()
```
Comment (longjiquan) [inline internal/rootcoord/meta_table.go:722]: Use `RLock` instead.

File: internal/proxy/impl.go [hunk 13]
```diff
@@ -2774,12 +2774,143 @@ func (node *Proxy) CreateAlias(ctx context.Context, request *milvuspb.CreateAlia
...
+	if !node.checkHealthy() {
+		return &milvuspb.DescribeAliasResponse{
+			Status: merr.Status(merr.WrapErrServiceNotReady(fmt.Sprintf("proxy %d is unhealthy", node.session.ServerID))),
+		}, nil
+	}
+
+	ctx, sp := otel.Tracer(typeutil.ProxyRole).Start(ctx, "Proxy-DescribeAlias")
+	defer sp.End()
+
+	dat := &DescribeAliasTask{
```
Comment (longjiquan) [inline internal/proxy/impl.go:2788]: How about using `return node.rootCoord.DescribeAlias` directly? Anyway, these two apis will be scheduled by the rootcoord's queue.

File: internal/proxy/task.go [hunk 14]
```diff
@@ -1932,6 +1934,132 @@ func (a *AlterAliasTask) PostExecute(ctx context.Context) error {
 	return nil
 }
 
+// DescribeAliasTask is the task to describe alias
+type DescribeAliasTask struct {
```
Comment (longjiquan) [inline internal/proxy/task.go:1938]: Please split these two structs into two files if these tasks are really required.

File: internal/rootcoord/meta_table.go [hunk 15]
```diff
@@ -697,6 +700,47 @@ func (mt *MetaTable) AlterAlias(ctx context.Context, alias string, collectionNam
...
+	defer mt.ddLock.Unlock()
+
+	var collectionFilterID int64 = -1
+	if collectionName != "" {
+		if collectionID, OK := mt.collName2ID[collectionName]; OK {
+			collectionFilterID = collectionID
+		} else {
+			return nil, merr.WrapErrCollectionNotFound(collectionName)
+		}
+	}
```
Comment (longjiquan) [inline internal/rootcoord/meta_table.go:732]: Please check if the collection was availiable.

File: internal/rootcoord/root_coord.go [hunk 16]
```diff
@@ -1657,6 +1662,86 @@ func (c *Core) AlterAlias(ctx context.Context, in *milvuspb.AlterAliasRequest) (
 	return merr.Status(nil), nil
 }
 
+// DescribeAlias describe collection alias
+func (c *Core) DescribeAlias(ctx context.Context, in *milvuspb.DescribeAliasRequest) (*milvuspb.DescribeAliasResponse, error) {
+	if code, ok := c.checkHealthy(); !ok {
```
Comment (longjiquan) [inline internal/rootcoord/root_coord.go:1667]: Please add ut for this case. You can just look at how we test the `CreateCollection`, and you'll know it's very easy to do this.

File: internal/rootcoord/root_coord.go [hunk 17]
```diff
@@ -1657,6 +1662,86 @@ func (c *Core) AlterAlias(ctx context.Context, in *milvuspb.AlterAliasRequest) (
...
+		zap.String("db", in.GetDbName()),
+		zap.String("alias", in.GetAlias()))
+
+	if in.GetAlias() == "" {
+		return &milvuspb.DescribeAliasResponse{
+			Status: merr.Status(merr.WrapErrParameterMissing("alias", "no input alias")),
+		}, nil
+	}
+
+	collectionName, err := c.meta.DescribeAlias(ctx, in.GetAlias(), 0)
```
Comment (longjiquan) [inline internal/rootcoord/root_coord.go:1687]: I prefer to letting these two requests scheduled by the rootcoord instead of proxy.

**Review by longjiquan — CHANGES_REQUESTED** [review]

**Comment by wayblink** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

**Pull request closed without merging**